src/clusterfuzz/_internal/google_cloud_utils/storage.py

```diff
@@ -249,23 +248,37 @@ def list_blobs(self, remote_path, recursive=True, names_only=False):
     else:
       fields = None
 
-    iterator = bucket.list_blobs(
-        prefix=path, delimiter=delimiter, fields=fields)
-    for blob in iterator:
-      properties['bucket'] = bucket_name
-      properties['name'] = blob.name
-      properties['updated'] = blob.updated
-      properties['size'] = blob.size
-
-      yield properties
-
-    if not recursive:
-      # When doing delimiter listings, the "directories" will be in `prefixes`.
-      for prefix in iterator.prefixes:
-        properties['bucket'] = bucket_name
-        properties['name'] = prefix
+    iterations = 0
+    while True:
+      iterations += 1
+      iterator = bucket.list_blobs(
+          prefix=path, delimiter=delimiter, fields=fields)
+      for blob in iterator:
+        properties = {
+            'bucket': bucket_name,
+            'name': blob.name,
+            'updated': blob.updated,
+            'size': blob.size,
+        }
+
         yield properties
 
+      if not recursive:
+        # When doing delimiter listings, the "directories" will be in
+        # `prefixes`.
+        for prefix in iterator.prefixes:
+          properties = {
+              'bucket': bucket_name,
+              'name': prefix,
+          }
+          yield properties
+
+      next_page_token = iterator.next_page_token
+      if next_page_token is None:
+        break
+      if iterations % 50 == 0:
+        logs.error('Might be infinite looping.')
```

Comment:
Why this magic number?

Reply:
Is there a better heuristic for guessing we are in an infinite loop :-)
It's an arbitrary number that I don't think should be realistically reached.

Reply:
How many elements show up per page?
What 

The heuristic approach is reasonable, but it would be nice to lay out in the open how many files at most are expected to be in a bucket (build/corpus), and why 50 * items_per_page is a good upper bound.

Reply:
Having this upper threshold and counting during the iteration might also be more resilient, if GCP changes the pagination behavior under the hood and you get, let's say, 10x less items, this might break in a very subtle way.